broker-core/src/main/java/io/zeebe/broker/workflow/model/transformation/transformer/ServiceTaskTransformer.java

```diff
@@ -80,34 +82,57 @@ private void transformTaskHeaders(ServiceTask element, final ExecutableServiceTa
     final ZeebeTaskHeaders taskHeaders = element.getSingleExtensionElement(ZeebeTaskHeaders.class);
 
     if (taskHeaders != null) {
-      final DirectBuffer encodedHeaders = encode(taskHeaders);
-      serviceTask.setEncodedHeaders(encodedHeaders);
+      final List<ZeebeHeader> validHeaders =
+          taskHeaders
+              .getHeaders()
+              .stream()
+              .filter(this::isValidHeader)
+              .collect(Collectors.toList());
+
+      if (validHeaders.size() < taskHeaders.getHeaders().size()) {
+        LOG.warn("Found task headers with no key or value. Ignoring");
+      }
+
+      if (validHeaders != null && !validHeaders.isEmpty()) {
```

Comment:
In which case would `validHeaders` be null? Even if we filter an empty list, we still get an empty list, no?